app/routes.py

```diff
@@ -1,2 +1,325 @@
+
+    video.title = request_data["title"]
+    video.release_date = request_data["release_date"]
+    video.total_inventory = request_data["total_inventory"]
+
+    db.session.commit()
+
+    currently_rented_out = len(video.rentals)
+    total = video.total_inventory
+    available = total - currently_rented_out
+
+    response_body = {
+            "id": video.id,
+            "title": video.title,
+            "release_date": video.release_date,
+            "total_inventory": video.total_inventory,
+            "available_inventory": available
+    }
+
+    return make_response(response_body, 200)
+
+@video_bp.route("/<video_id>", methods=["DELETE"], strict_slashes=False)
+def delete_video(video_id):
+    video = Video.query.get(video_id)
+
+    if video is None:
+        return make_response({"details": f"Video id number: {video_id} not found"}, 404)
+
+    db.session.delete(video)
+    db.session.commit()
+
+    return make_response({"id": f"{video.id}"}, 200)
+
+
+#Rental routes and many to many routes
+
+@rental_bp.route("/check-out", methods=["POST"],strict_slashes=False)
+def check_out_video():
+
+    request_body = request.get_json()
+
+    customer_id = request_body["customer_id"]
+    video_id = request_body["video_id"]
+    customer = Customer.query.get(customer_id)
+    video = Video.query.get(video_id)
+
+
+    if not request_body:
+        return make_response("", 404)
+
+    if video is None:
+        return make_response({"details": f"Video id number: {video_id} not found"}, 404)
+    
+    if customer is None:
+        return make_response({"details": f"Video id number: {customer_id} not found"}, 404)
+
+    now = datetime.now()
+    due = timedelta(days=+7)
+    due_date = now + due   
+
+    new_rental = Rental(
+        customer_id = request_body["customer_id"],
+        video_id = request_body["video_id"], 
+        due_date = due_date
+
+    )
+
+    video.total_inventory = video.total_inventory - 1
+
+    db.session.add(new_rental)
+    db.session.commit()
+
+    checked_out_count = len(customer.rentals)
+
+    response_body = {
+            "customer_id": customer_id,
+            "video_id": video_id,
+            "due_date": new_rental.due_date,
+            "videos_checked_out_count": checked_out_count,
+            "available_inventory": video.total_inventory
+    }
+
+    return make_response(response_body, 200)
+
+@rental_bp.route("/check-in", methods=["POST"],strict_slashes=False)
+def check_in_video():
+
+    request_body = request.get_json()
+
+    if not request_body:
+        return make_response("", 404)
+
+    customer_id = request_body["customer_id"]
+    video_id = request_body["video_id"]
+    customer = Customer.query.get(customer_id)
+    video = Video.query.get(video_id)
+
+    Rental.query.filter_by(customer_id=customer_id, video_id=video_id).delete()
+
+    video.total_inventory = video.total_inventory + 1
+    
+    db.session.commit()
+
+    checked_out_count = len(customer.rentals)
```

Comment:
This also makes a good thing to move into a model function.

Lastly your deleting of rentals works as far as the purpose we need, but if you ever wanted to get the records for who has rented a video in the past... it's gone.